docs/tpm2-pkcs11_object_auth_model.md

```diff
@@ -0,0 +1,528 @@
+* The TOKEN ultimately protects all the USER created objects as well as access
+through an authorization model to distinguish the SO, USER or adversary.
+* USER objects can be keys, certificates or any user defined data.
+
+## PKCS11 Roles
+Overview of pkcs11 requirements of the security officer and user roles. The
+PKCS11 design describes two specific roles - A security-officer or **SO**
+and **USER**. These roles are realized with separate PIN authentications.
+Following are some relevant design considerations when comparing the roles:
+* Neither SO nor USER can know each other's PIN.
+* The SO is restricted to an administrative role to modify SO PIN and or an
+existing USER's PIN.
+* While both the **SO and the USER can change USER PIN**, only a SO can change
+SO PIN.
+* USER authentication PIN is a must for operating on USER objects.
+* The SO cannot access any USER objects.
+* While SO could change USER PIN, the **SO cannot restore the old USER PIN**.
+This intentionally prevents SO from having explicit access to USER objects and
+also to bring traceability to security officer actions regarding PIN changes for
+USER authorization.
+<br>
+As an example, a token initialized with a SO PIN can be setup with a default
+USER PIN. The USER then changes the PIN to his liking which the SO cannot know
+& thus cant restore. The USER can now use the new PIN to create and operate on
+USER objects. If the USER is convinced of forgetting the PIN, USER can get the
+PIN reset by the SO. However, if the SO resets the USER PIN without USER
+request, the old USER PIN cannot be restored or used; thus revealing the SO act
+of changing the USER PIN.
+
+## TPM2 Objects
+Overview of the hierarchical concept of objects.
+The TPM2 standards defines various types of objects - permanent, transient,
+persistent, non-volatile or NV, etc. There are TPM entities that also have a
+parent:child relationship and hence embody the intended hierarchical
+relationship defined by PKCS11. This hierarchical relationship can be formed
+by object and authorization policy definitions. In general, parent authorization
+is required to be able to authorize the child object.
+
+### The TPM hierarchy
+This is the parent node of all the possible child objects to be created. There
+are 4 hierarchies - Platform, Owner, Endorsement and NULL hierarchy. Only two of
+these can authorize creating and managing of NV objects - Those are Platform and
+Owner hierarchy. NV objects include persistent and generic NV indexes. Hierarchy
+authorization is also required for Creating primary objects.
+
+### The Primary Object
+Except for NULL hierarchy primary objects under all other hierarchies can be
+recreated after a TPM2 reset. Hierarchy authorization is required every time a
+primary object needs to be created. Once the primary object is created it can be
+made persistent. A persistent and password-less primary object does not require
+any further authorization to use even after TPM2 reset.
```

Comment:
Generally people make these a well known auth value of size 0 (no auth) or more. I've seen things like 123456. So I would add either passwordless or with a well known auth value.